x/merkledb/key.go

```diff
@@ -11,109 +11,124 @@ import (
 )
 
 var (
-	errInvalidBranchFactor = errors.New("invalid branch factor")
-
-	branchFactorToTokenConfig = map[BranchFactor]tokenConfig{
-		BranchFactor2: {
-			branchFactor:    BranchFactor2,
-			tokenBitSize:    1,
-			tokensPerByte:   8,
-			singleTokenMask: 0b0000_0001,
-		},
-		BranchFactor4: {
-			branchFactor:    BranchFactor4,
-			tokenBitSize:    2,
-			tokensPerByte:   4,
-			singleTokenMask: 0b0000_0011,
-		},
-		BranchFactor16: {
-			branchFactor:    BranchFactor16,
-			tokenBitSize:    4,
-			tokensPerByte:   2,
-			singleTokenMask: 0b0000_1111,
-		},
-		BranchFactor256: {
-			branchFactor:    BranchFactor256,
-			tokenBitSize:    8,
-			tokensPerByte:   1,
-			singleTokenMask: 0b1111_1111,
-		},
+	ErrInvalidTokenConfig = errors.New("token configuration must match one of the predefined configurations ")
+
+	BranchFactor2TokenConfig = TokenConfiguration{
+		branchFactor:    2,
+		tokenBitSize:    1,
+		tokensPerByte:   8,
+		singleTokenMask: 0b0000_0001,
+	}
+	BranchFactor4TokenConfig = TokenConfiguration{
+		branchFactor:    4,
+		tokenBitSize:    2,
+		tokensPerByte:   4,
+		singleTokenMask: 0b0000_0011,
+	}
+	BranchFactor16TokenConfig = TokenConfiguration{
+		branchFactor:    16,
+		tokenBitSize:    4,
+		tokensPerByte:   2,
+		singleTokenMask: 0b0000_1111,
+	}
+	BranchFactor256TokenConfig = TokenConfiguration{
+		branchFactor:    256,
+		tokenBitSize:    8,
+		tokensPerByte:   1,
+		singleTokenMask: 0b1111_1111,
+	}
+	validTokenConfigurations = []TokenConfiguration{
+		BranchFactor2TokenConfig,
+		BranchFactor4TokenConfig,
+		BranchFactor16TokenConfig,
+		BranchFactor256TokenConfig,
 	}
 )
 
-type BranchFactor int
-
-const (
-	BranchFactor2   BranchFactor = 2
-	BranchFactor4   BranchFactor = 4
-	BranchFactor16  BranchFactor = 16
-	BranchFactor256 BranchFactor = 256
-)
+type TokenConfiguration struct {
+	branchFactor    int
+	tokensPerByte   int
+	tokenBitSize    int
+	singleTokenMask byte
```

Comment:
Should we export these and remove the same-named methods?

Reply:
I don't want them to be editable